src/Microsoft.Data.Analysis/ColumnArithmeticTemplate.ttinclude

```diff
@@ -88,6 +120,21 @@
         return false;
     }
 
+    // These are a subset of the implicit conversions allowed in C#. They are used to generate the return type for binary ops
+    public Dictionary<string, LinkedList<string>> primitiveTypeToImplicitConversions = new Dictionary<string, LinkedList<string>> {
+        {"sbyte", new LinkedList<string>(new List<string> {"int", "long", "float", "double", "decimal"})},
+        {"byte", new LinkedList<string>(new List<string> {"int", "uint", "long", "ulong", "float", "double", "decimal"})},
```

Comment:
1. you are missing `short` in these.
2. The way you have `GetBinaryOperationReturnType` implemented, you will need the actual type at the front of this list too. The way you have it today, when you add 2 `byte` columns, you are returning an `int` column. It should be returning a `byte`.

Reply:
I just looked at the compiler's behavior, and adding 2 bytes returns an `Int32` in C#. So I think I'm wrong here.

---
In reply to: [395296638](https://github.com/dotnet/corefxlab/pull/2867#discussion_r395296638) [](ancestors = 395296638)